### DISCUSSION TITLE
JK-B2A24S20P - RS485 Connector Partnumber

### DISCUSSION THREAD

**Discussion by ircoopr** [syssi/esphome-jk-bms#136] [General]
Hello all,

Fantastic work on this release (and the CANbus enabled fork) - it's really helped speed up my development!

It looks like plenty of people have got this working, my query is how are people connecting in to the RS485 port? I know that we don't need to use the JK dongle since we only need TTL level signals for direct connection to an ESP - I don't really want to have to buy the dongle just to get the connector. Does anyone have a part number for the connector that is used?

Additional, I haven't found it super clear which port exactly is RS485 - is it the one that's labelled GPS? - my unit seems to how CAN/RS485 on the left most connector (printed on the case) - but then the diagrams show that RS485 is on the second connector along (printed GPS on the case).

Thanks!
Ian

**Comment by syssi** [comment]
The type of the connector is a bit hidden at the docs: https://github.com/syssi/esphome-jk-bms#schematics

> # RS485-TTL jack (4 Pin, JST 1.25mm pitch)

This is the port you are looking for: https://github.com/syssi/esphome-jk-bms/discussions/119#discussioncomment-3016399

**Reply by ircoopr** [reply]
Thanks, sorry I should have looked harder! Thanks again!

**Reply by syssi** [reply]
I will move the info to a more prominent location. You are not the first one who missed the detail.

**Reply by syssi** [reply]
Another detail which is hard to spot: The RS485-TTL connection is **read-only**. You cannot change settings or control the mosfet states using the serial connection.

**Comment by linuxdep** [comment]
Warum nutzt ihr den RS485 Adapter nicht? habe diesen da, da dort auch der Stecker schon dran ist, kann der genau so benutzt werden? Sollte ja eigentlich oder passt da der UART TTL Software-Treiber dann nicht? Dachte eigentlich der RS485 Adapter setzt gerade von TTL UART auf RS485 um, das im BMS verbaut ist?

**Reply by syssi** [reply]
The Jikong RS485 dongle converts the UART-TTL signal from the BMS into a differential RS485 signal. If you want to use the dongle you have to convert the RS485 signal back to UART-TTL again with s second converter module:

```
BMS<--->JK-RS485-Dongle<--->RS485-Module<--->ESP
```

**Reply by linuxdep** [reply]
OK, the BMS has UART TTL 3,3V or 5V? I thought the BMS has RS485 and the adapter converts it back, then it's clear. So I immediately ordered a plug with cable... but first test with the BLE if it works reliably.

**Reply by syssi** [reply]
You can measure the logic level easily if you measure the voltage between TX and GND. It's 3.3V.
